src/ray/common/function_descriptor.h

```diff
@@ -120,6 +129,12 @@ class JavaFunctionDescriptor : public FunctionDescriptorInterface {
            ", signature=" + typed_message_->signature() + "}";
   }
 
+  virtual std::string CallString() const {
+    std::string class_name = typed_message_->class_name();
+    std::string function_name = typed_message_->function_name();
+    return class_name.empty() ? function_name : class_name + "." + function_name;
```

Comment:
@rkooo567 Could you elaborate on your reasoning for this change? I broke out `class_name` and `function_name` since this seemed more readable than
```c++
return typed_message_->class_name().empty() ? typed_message_->function_name() : typed_message_->class_name() + "." + typed_message_->function_name();
```
and would eliminate two unnecessary proto message accessor function calls. What you've suggested seems to keep the `class_name` and `function_name` variables, but still duplicates the proto message accessor function calls.

Reply:
oh in this case, I meant to delete the first two lines (it seems like it is same for other methods). Sorry for confusion. Also, this comment is not a hard requirement. (I initially thought it can save one copy constructor, but behavior might be the same. I am not 100% sure... cpp world :)).

Reply:
Ah that makes more sense! I'll double-check that C++ behavior.

Reply:
Yeah I am not sure if it is true because I am also not a cpp expert. I think it can be actually just the same. Let me know what's the right behavior.

Reply:
We can keep the readability and the elimination of the two extra proto message accessor function calls by changing the `std::string`s to const references, which is what is returned by the [C++ generated proto code for message field accessors](https://developers.google.com/protocol-buffers/docs/reference/cpp-generated#proto3_string). Only the necessary copies, in either returning a copy of `function_name` or constructing the concatenated class name + function name, should be made.